.github/files/update-project-mirrors.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+
+git_setup()
+{
+  cat <<- EOF > "$HOME/.netrc"
+		machine github.com
+			login matticbot
+			password $GITHUB_TOKEN
+		machine api.github.com
+			login matticbot
+			password $GITHUB_TOKEN
+EOF
+  chmod 600 "$HOME/.netrc"
+
+  git config --global user.email "matticbot@users.noreply.github.com"
+  git config --global user.name "matticbot"
+}
+
+# Halt on error
+set -eo pipefail
+
+if [[ -n "$CI" ]]; then
+	git_setup
+fi
+
+BASE=$(pwd)
+CLONE_BASE=$(mktemp -d "${TMPDIR:-/tmp}/jetpack-update-project-mirrors.XXXXXXXX")
+MONOREPO_COMMIT_MESSAGE=$(git show -s --format=%B $GITHUB_SHA)
+COMMIT_MESSAGE=$( echo "${MONOREPO_COMMIT_MESSAGE}\n\nCommitted via a GitHub action: https://github.com/automattic/jetpack/runs/${GITHUB_RUN_ID}" )
+COMMIT_ORIGINAL_AUTHOR="${GITHUB_ACTOR} <${GITHUB_ACTOR}@users.noreply.github.com>"
+
+if [[ "$GITHUB_REF" =~ ^refs/heads/ ]]; then
+	BRANCH=${GITHUB_REF#refs/heads/}
+else
+	echo "Could not determine branch name from $GITHUB_REF"
+	exit 1
+fi
+
+# Install Yarn generally.
+echo "::group::Monorepo setup"
+yarn install
+echo "::endgroup::"
+
+echo "Cloning projects and pushing to Automattic mirror repos"
+EXIT=0
+
+# sync to read-only clones
+for project in projects/packages/* projects/plugins/*; do
+	PROJECT_DIR="${BASE}/${project}"
```

Comment:
should we worry about any naming collisions here? A plugin and a package both called `jetpack-cdn` or something to that effect?

Reply:
Shouldn't be a problem.

* `$project` would be `projects/packages/cdn` and `projects/plugins/cdn`.
* `$PROJECT_DIR` would similarly have the full path.
* Hopefully they don't have the same `name` in their composer.json files.